### PR TITLE
feat: give maintainers access to the EKS cluster TDE-926

### PR DESCRIPTION
**Pull request by paulfouquet**
#### Motivation

Maintainers should have direct access to the EKS Cluster.

#### Modification

Pass the maintainer role ARNs as a CDK context argument so they can be added as master on the cluster

#### Checklist

_If not applicable, provide explanation of why._

- [x] Tests updated
- [ ] Docs updated
- [x] Issue linked in Title

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,8 @@
   },
   "scripts": {
     "lint": "npx eslint . --ignore-path .gitignore",
-    "format": "npx prettier . -w"
+    "format": "npx prettier . -w",
+    "test": "node --loader tsx --test infra/**/*.test.ts"
```
Comment (blacha) [inline package.json:22]: does this need to be restricted to `infra/` does it work without it?

File: infra/eks/cluster.ts [hunk 2]
```diff
@@ -71,14 +72,11 @@ export class LinzEksCluster extends Stack {
...
-    this.cluster.awsAuth.addMastersRole(accountAdminRole);
-
-    // If defined allow the CI user access to the cluster
-    if (props.ciRoleArn) {
-      const ciRole = Role.fromRoleArn(this, 'CiRole', props.ciRoleArn);
-      this.cluster.awsAuth.addMastersRole(ciRole);
+    //allow the maintainer roles access to the cluster
+    for (const roleArn of props.maintainerRoleArns) {
+      const roleId = createHash('sha256').update(roleArn).digest('hex').slice(0, 12);
+      const role = Role.fromRoleArn(this, roleId, roleArn);
```
Comment (blacha) [inline infra/eks/cluster.ts:78]: should have some sort of prefix about the id, and needs a default policy doc

`AdminRole-${roleId}` ?

File: infra/eks/arn.ts [hunk 3]
```diff
@@ -42,8 +42,11 @@ export function tryGetContextArn(node: Node, context: string): string | null {
  * @returns arns if they are valid, null otherwise
  */
 export function tryGetContextArns(node: Node, context: string): string[] | null {
-  const ctx = node.tryGetContext(context);
+  let ctx = node.tryGetContext(context);
   if (ctx == null) return null;
```
Comment (l0b0) [inline infra/eks/arn.ts:46]: We could possibly return `[]` in this case, to avoid extra type handling.

File: package.json (hunk 1 above)
Comment (paulfouquet) [reply package.json:22]: I can't make it work without specifying `infra`

Comment (blacha) [reply package.json:22]: what about `... --test infra/` I think it should be looking for all `.test.ts` maybe its not looking for ts files.

Comment (blacha) [reply package.json:22]: hmm maybe not, I think ** doesnt work in windows but :shrug: we don't build in windows.

File: infra/eks/arn.ts (hunk 3 above)
Comment (blacha) [reply infra/eks/arn.ts:46]: null is more common, as you would still need to do `if (ret.length === 0) throw error`

Using `?` means it you can generally ignore the fact that it returned null eg

```typescript
const arns = tryGetContextArns(...)
const arn = arns?.find(f => f.includes("internal-user"))
if (arn == null) throw new Error("Cannot find internal-user role")
```

File: package.json (hunk 1 above)
Comment (l0b0) [reply package.json:22]: Probably `**` could be just `*` - most contexts don't support recursive globs.

Comment (paulfouquet) [reply package.json:22]: I can change it to `*` if needed

Comment (blacha) [reply package.json:22]: I would leave as is, as `*` wouldn't cover `infra/foo/bar/test.ts` a more complex option would be to use `find` or a third party glob eg `glob -c ".." "**/.test.ts"`

**Review by l0b0 — APPROVED** [review]

**Review by blacha — APPROVED** [review]

**Pull request merged**